topydo/ui/columns/Transaction.py

```diff
@@ -0,0 +1,76 @@
+# Topydo - A todo.txt client written in Python.
+# Copyright (C) 2017 Bram Schoenmakers <bram@topydo.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from topydo.lib.MultiCommand import MultiCommand
+
+
+class Transaction(object):
+    """
+    This class implements basic handling of executing any subcommand on multiple
+    todo items.
+    """
+    def __init__(self, p_subcommand=None, p_env_args=(), p_todo_ids=None):
+        self._multi = issubclass(p_subcommand, MultiCommand)
+        self._cmd = lambda op: p_subcommand(op, *p_env_args)
+        self._todo_ids = p_todo_ids
+        self._operations = []
+        self._cmd_name = p_subcommand.__module__.lower()[16:-7]
```

Comment:
This is a duplicate of [line 252 of CLIApplicationBase.py](https://github.com/mruwek/topydo/blob/c1fe991ac4942b80b238e05eb9264c06ca0f01d0/topydo/ui/CLIApplicationBase.py#L252). Since it's a bit obscure (range 16:-7), it deserves to be refactored to a separate function with a descriptive name (perhaps it should be a method of the `Command` base class).